src/components/Common/TokenDecimals.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import { validateDecimals } from '../../utils/validations'
+import { TEXT_FIELDS } from '../../utils/constants'
+import { Field } from 'react-final-form'
+import { InputField2 } from './InputField2'
+
+export const TokenDecimals = ({ disabled, errorStyle }) => (
+  <Field
+    validate={validateDecimals}
+    component={InputField2}
+    parse={(value) => /^[0-9]*$/.test(value) ? value : ''}
```

Comment:
This causes the whole input to be cleared if some typed character is invalid.

Reply:
Yes, just refactored. Now works as expected.